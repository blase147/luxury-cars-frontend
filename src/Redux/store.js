import { configureStore } from '@reduxjs/toolkit';
import authReducer from './Reducers/authSlice';
import carsReducer from './Reducers/carSlice';
import carReducer from './Reducers/carDetailsSlice';
import signUpReducer from './Reducers/regSlice';

const store = configureStore({
  reducer: {
    auth: authReducer,
    register: signUpReducer,
    houses: carsReducer,
    houseDetail: carReducer,
  },
});

export default store;
