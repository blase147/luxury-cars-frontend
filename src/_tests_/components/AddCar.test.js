import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AddCar from '../../components/addCars/addCar';
import store from '../../Redux/store';

describe('It renders', () => {
  it('Adds car correctly', () => {
    const tree = render(
      <Provider store={store}>
        <MemoryRouter>
          <AddCar />
        </MemoryRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
