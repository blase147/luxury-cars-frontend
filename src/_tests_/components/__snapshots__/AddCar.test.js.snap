// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders Adds car correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <form>
          <h2>
            ADD LUXURY CARS
          </h2>
          <div
            class="inputEle first"
          >
            <input
              name="name"
              placeholder="Car Name"
              type="text"
            />
          </div>
          <div
            class="inputEle"
          >
            <input
              name="brand"
              placeholder="Car brand"
              type="text"
            />
          </div>
          <div
            class="inputEle"
          >
            <input
              name="image"
              placeholder="Car ImageUrl"
              type="text"
            />
          </div>
          <div
            class="inputEle"
          >
            <input
              name="lending_fee"
              placeholder="Car lending_fee"
              type="number"
            />
          </div>
          <div
            class="inputEle"
          >
            <textarea
              id="message"
              name="description"
              placeholder="Car Description..."
              rows="4"
            />
          </div>
          <div
            class="inputEle submit"
          >
            <button
              class="btnsubmitf"
              type="button"
            >
              Add Car
            </button>
          </div>
          <a
            href="/"
          >
            <button
              aria-label="Go back"
              class="returnBtn"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M321.94 98L158.82 237.78a24 24 0 000 36.44L321.94 414c15.57 13.34 39.62 2.28 39.62-18.22v-279.6c0-20.5-24.05-31.56-39.62-18.18z"
                />
              </svg>
            </button>
          </a>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <form>
        <h2>
          ADD LUXURY CARS
        </h2>
        <div
          class="inputEle first"
        >
          <input
            name="name"
            placeholder="Car Name"
            type="text"
          />
        </div>
        <div
          class="inputEle"
        >
          <input
            name="brand"
            placeholder="Car brand"
            type="text"
          />
        </div>
        <div
          class="inputEle"
        >
          <input
            name="image"
            placeholder="Car ImageUrl"
            type="text"
          />
        </div>
        <div
          class="inputEle"
        >
          <input
            name="lending_fee"
            placeholder="Car lending_fee"
            type="number"
          />
        </div>
        <div
          class="inputEle"
        >
          <textarea
            id="message"
            name="description"
            placeholder="Car Description..."
            rows="4"
          />
        </div>
        <div
          class="inputEle submit"
        >
          <button
            class="btnsubmitf"
            type="button"
          >
            Add Car
          </button>
        </div>
        <a
          href="/"
        >
          <button
            aria-label="Go back"
            class="returnBtn"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M321.94 98L158.82 237.78a24 24 0 000 36.44L321.94 414c15.57 13.34 39.62 2.28 39.62-18.22v-279.6c0-20.5-24.05-31.56-39.62-18.18z"
              />
            </svg>
          </button>
        </a>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
